include/dgl/zerocopy_serializer.h

```diff
@@ -0,0 +1,112 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file rpc/shared_mem_serializer.h
+ * \brief headers for serializer.
+ */
+#ifndef DGL_ZEROCOPY_SERIALIZER_H_
+#define DGL_ZEROCOPY_SERIALIZER_H_
+
+#include <dgl/runtime/ndarray.h>
+#include <dmlc/io.h>
+#include <dmlc/memory_io.h>
+#include <dmlc/serializer.h>
+
+#include <deque>
+#include <queue>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "dmlc/logging.h"
+
+namespace dgl {
+
+// StringStreamWithBuffer is backed up by a string. It stores the data pointer
+// seperately instead of directly write in to the stream.
```

Comment:
```
// ... . The class supports serializing and deserializing NDArrays stored in shared memory.
// If the stream is created for sending/recving data through network, the data pointer of
// the NDArray will be transmitted directly without and copy. Otherwise, the stream is for
// sending/recving data to another process on the same machine, so if an NDArray is
// stored in shared memory, it will just record the shared memory name instead of the
// actual data buffer.
```